crates/astria-grpc-mock/src/response.rs

```diff
@@ -54,6 +56,52 @@ impl Respond for DefaultResponse {
     }
 }
 
+pub fn dynamic_response<I, O, F>(responder: F) -> DynamicResponse<I, O, F>
```

Comment:
I wonder if we can implement e.g.:
```rust
impl<I, O, F> Respond for F,
where
    F: Fn(&I) -> O,
    O: erased_serde::Serialize + prost::Name + Clone + 'static,
```
Maybe another indirection would be necessary like `MakeDynamicResponse` and then implemeent `Respond for MakeDynamicResponse`. Gonna think about how to do that.

Reply:
Yeah - that was my initial approach, but `I` and `O` are unconstrained type parameters so it doesn't fly :)

I played with this for a while but the explicit `dynamic_response` function seemed like the path of least resistance.  Maybe there's a cleaner way though?